packages/gateway/src/middleware/mw_ip_filtering.ts

```diff
@@ -0,0 +1,63 @@
+import { RequestHandler } from 'express';
+import { ApiDef } from '@graphql-portal/types';
+import { RequestMiddleware } from './request-middleware.interface';
+import { prefixLogger } from '@graphql-portal/logger';
+import { getClientIp } from 'request-ip';
+import { inRange, isIP } from 'range_check';
+
+type Strategy = 'allow' | 'deny' | 'none';
+
+const getMiddleware: RequestMiddleware = function (apiDef: ApiDef): RequestHandler {
+  // skipping if disabled
+  if (apiDef.enable_ip_filtering !== true) {
+    return (req, res, next) => next();
+  }
+
+  let ips: string[] = [];
+  let strategy: Strategy = 'none';
+  const logger = prefixLogger('mw_ip_blacklist');
+  const ipValidator = function (ip: string): boolean {
+    if (!isIP(ip)) {
+      logger.warn('Incorrect IP: "%s". Skipping...', ip);
+      return false;
+    }
+    return true;
+  };
+
+  if (apiDef.allow_ips && apiDef.allow_ips.length > 0) {
+    strategy = 'allow';
+    ips = apiDef.allow_ips.filter(ipValidator);
+    logger.info('Enabling requests from the following IPs: %s', ips);
+  }
+
+  if (apiDef.deny_ips && apiDef.deny_ips.length > 0 && strategy != 'allow') {
```

Comment:
the same

Reply:
strategy != 'allow', so it cannot work as a white and black list at the same time?

Reply:
> strategy != 'allow', so it cannot work as a white and black list at the same time?

exactly